src/main/java/com/iemr/tm/utils/JwtUserIdValidationFilter.java

```diff
@@ -0,0 +1,109 @@
+
+@Component
+public class JwtUserIdValidationFilter implements Filter {
+
+	private final JwtAuthenticationUtil jwtAuthenticationUtil;
+	private final Logger logger = LoggerFactory.getLogger(this.getClass().getName());
+
+	public JwtUserIdValidationFilter(JwtAuthenticationUtil jwtAuthenticationUtil) {
+		this.jwtAuthenticationUtil = jwtAuthenticationUtil;
+	}
+
+	@Override
+	public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+			throws IOException, ServletException {
+		HttpServletRequest request = (HttpServletRequest) servletRequest;
+		HttpServletResponse response = (HttpServletResponse) servletResponse;
+
+		String path = request.getRequestURI();
+		String contextPath = request.getContextPath();
+		logger.info("JwtUserIdValidationFilter invoked for path: " + path);
+
+		// Log cookies for debugging
+		Cookie[] cookies = request.getCookies();
+		if (cookies != null) {
+			for (Cookie cookie : cookies) {
+				if ("userId".equals(cookie.getName())) {
+					logger.warn("userId found in cookies! Clearing it...");
+					clearUserIdCookie(response); // Explicitly remove userId cookie
+				}
+			}
+		} else {
+			logger.info("No cookies found in the request");
+		}
+
+		// Log headers for debugging
+		String jwtTokenFromHeader = request.getHeader("Jwttoken");
+		logger.info("JWT token from header: ");
+
+		// Skip login and public endpoints
+		if (path.equals(contextPath + "/user/userAuthenticate")
+				|| path.equalsIgnoreCase(contextPath + "/user/logOutUserFromConcurrentSession")
+				|| path.startsWith(contextPath + "/public")) {
+			logger.info("Skipping filter for path: " + path);
+			filterChain.doFilter(servletRequest, servletResponse);
+			return;
+		}
+
+		try {
+			// Retrieve JWT token from cookies
+			String jwtTokenFromCookie = getJwtTokenFromCookies(request);
+			logger.info("JWT token from cookie: ");
+
+			// Determine which token (cookie or header) to validate
+			String jwtToken = jwtTokenFromCookie != null ? jwtTokenFromCookie : jwtTokenFromHeader;
+			if (jwtToken == null) {
+				response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "JWT token not found in cookies or headers");
+				return;
+			}
+
+			// Validate JWT token and userId
+			boolean isValid = jwtAuthenticationUtil.validateUserIdAndJwtToken(jwtToken);
+
+			if (isValid) {
+				// If token is valid, allow the request to proceed
+				filterChain.doFilter(servletRequest, servletResponse);
+			} else {
+				response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Invalid JWT token");
+			}
+		} catch (Exception e) {
+			logger.error("Authorization error: ", e);
+			response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Authorization error: " + e.getMessage());
+		}
+	}
+
+	private String getJwtTokenFromCookies(HttpServletRequest request) {
+		Cookie[] cookies = request.getCookies();
+		if (cookies != null) {
+			for (Cookie cookie : cookies) {
+				if (cookie.getName().equals("Jwttoken")) {
```

Comment:
Can you use JwtToken @indraniBan ?
With capital T

Reply:
From ui side cookie key name does not support camel case, either we have to use Jwt-Token or as it is